torch/csrc/CudaIPCTypes.cpp

```diff
@@ -0,0 +1,156 @@
+#ifdef USE_CUDA
+#include <torch/csrc/CudaIPCTypes.h>
+#include <map>
+#include <mutex>
+
+namespace torch {
+
+void BadIPCPatternError() {
+  static bool warned = false;
+  if (not warned) {
+    LOG(WARNING)
+        << "Producer process has been terminated before all shared CUDA tensors released.";
```

Comment:
This message can be *even better* if it offers some information about what this means, and advice about how to remediate the situation. A link to more detailed docs is often good enough.